app/controllers/shelters_controller.rb

```diff
@@ -23,10 +28,17 @@ def edit
   def update
     @shelter = Shelter.find(params[:id])
     @shelter.update(shelter_params)
-    redirect_to "/shelters/#{@shelter.id}"
+    if @shelter.save
+      redirect_to "/shelters/#{@shelter.id}"
+    else
+      flash[:notice] = "Unable to update shelter: Required fields are empty."
+      render :edit
+    end
   end
 
   def destroy
+    shelter = Shelter.find(params[:id])
+    shelter.destroy_associated
```

Comment:
Nice - so this will destroy everything in the shelter, and not just the shelter itself